libs/stats/odc/stats/model.py

```diff
@@ -234,6 +238,17 @@ def extend_if_needed(self, seconds, buffer_seconds: int = 30) -> bool:
             return self.extend(seconds)
         return True
 
+    def register_heartbeat(self):
```

Comment:
This is not the right place for this. It should be TaskRunner concern, also location of the file should be configurable, we might be running several workers on the same machine.